### PR TITLE
Beam integration

### PR DESCRIPTION
**Pull request by tiopi**
Beam integration requires #439. New files include `beam.py`, `test_beam.py` and test additions to tox.

### CODE REVIEW

File: sentry_sdk/integrations/beam.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+        ignore_logger("root")
+        ignore_logger("bundle_processor.create")
+
+        function_patches = ["process", "start_bundle", "finish_bundle", "setup"]
+        for func_name in function_patches:
+            setattr(
+                DoFn,
+                INSPECT_FUNC.format(func_name),
+                patched_inspect_process(DoFn, func_name),
```
Comment (Zylphrex) [inline sentry_sdk/integrations/beam.py]: this no longer just patch the `_inspect_process` method, consider renaming it

File: sentry_sdk/client.py [hunk 2]
```diff
@@ -74,23 +74,37 @@ class _Client(object):
     """
 
     def __init__(self, *args, **kwargs):
-        # type: (*Optional[str], **Any) -> None
+        # type: (*Any, **Any) -> None
```
Comment (untitaker) [inline sentry_sdk/client.py]: Not sure what happened here but this diff seems to be the result of merge conflict resolution and not actually intended.

File: sentry_sdk/integrations/beam.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        client = hub.client
+        event, hint = event_from_exception(
+            exc_info,
+            client_options=client.options,
+            mechanism={"type": "beam", "handled": False},
+        )
+        hub.capture_event(event, hint=hint)
+
+
+def raiseException(client):
```
Comment (untitaker) [inline sentry_sdk/integrations/beam.py]: For consistency, could you use snake_case here as well?

File: tests/test_client.py [hunk 4]
```diff
@@ -359,28 +350,6 @@ def callback(scope):
     assert not calls
 
 
-@pytest.mark.parametrize("debug", (True, False))
```
Comment (untitaker) [inline tests/test_client.py]: This seems like a botched merge as well

**Pull request merged**